packages/xo-web/src/xo-app/new/sr/index.js

```diff
@@ -313,17 +376,39 @@ export default class New extends Component {
 
   _handleSrTypeSelection = event => {
     const type = event.target.value
+    if (type === 'hba') {
+      this._handleSrHbaSelection()
+    }
     this.setState({
-      type,
-      paths: undefined,
+      hbaDevices: undefined,
       iqns: undefined,
+      paths: undefined,
+      summary: type === 'lvm' || type === 'local' || type === 'smb',
+      type,
+      unused: undefined,
       usage: undefined,
       used: undefined,
-      unused: undefined,
-      summary: type === 'lvm' || type === 'local' || type === 'smb',
     })
   }
 
+  _handleSrHbaSelection = async scsiId => {
+    const { host } = this.state
+
+    try {
+      this.setState(({ loading }) => ({ loading: loading + 1, scsiId }))
+      const hbaDevices = await probeSrHba(host.id)
```

Comment:
`_handleSrHbaSelection` should either be triggered when you select the `HBA` type or when you select an HBA SR in the list. But here, as you're using it for both of the purposes, you're probing for HBA SRs everytime you select an option, potentially messing with the `SelectScsiId`'s state.
You'd need to use this method only as a change handler and do the probing stuff in `_handleSrTypeSelection`.